models/DataObject/Concrete.php

```diff
@@ -745,17 +745,14 @@ public function __wakeup()
     {
         parent::__wakeup();
 
-        // renew localized fields
-        // do not use the getter ($this->getLocalizedfields()) as it somehow slows down the process around a sec
-        // no clue why this happens
-        if (property_exists($this, 'localizedfields') && $this->localizedfields instanceof Localizedfield) {
-            $this->localizedfields->setObject($this, false);
-        }
-
         // renew object reference to other object aware fields
         foreach ($this->__objectAwareFields as $objectAwareField => $exists) {
             if (isset($this->$objectAwareField) && $this->$objectAwareField instanceof ObjectAwareFieldInterface) {
-                $this->$objectAwareField->setObject($this);
+                if($this->$objectAwareField instanceof Localizedfield){
```

Comment:
lets create another interface `ObjectAwareFieldWithMarkAsDirtyInterface` that extends from the `ObjectAwareFieldInterface` and use the interface check in that `if` instead of the `Localizedfield` check

`Localizedfield` then should implement the `ObjectAwareFieldWithMarkAsDirtyInterface`

Reply:
done 👍